File: go-client-sdk/pkg/models/shared/codatassessdatacontractsaccountcategoriesaccountcategoriesmodelpagedresponsemodel.go
package shared

// CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseModel struct {
	Links        *CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                                                    `json:"pageNumber,omitempty"`
	PageSize     *int                                                                                    `json:"pageSize,omitempty"`
	Results      []CodatAssessDataContractsAccountCategoriesAccountCategoriesModel                       `json:"results,omitempty"`
	TotalResults *int                                                                                    `json:"totalResults,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatassessdatacontractsaccountcategoriesaccountcategoriesmodelpagedresponsemodel.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseLinksModel } from "./codatassessdatacontractsaccountcategoriesaccountcategoriesmodelpagedresponselinksmodel";
import { CodatAssessDataContractsAccountCategoriesAccountCategoriesModel } from "./codatassessdatacontractsaccountcategoriesaccountcategoriesmodel";



// CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseModel
/** 
 * Used to represent what can be returned by an endpoint that supports paging.
 * Usable with the [ProducesResponseType] attribute on a controller action.
**/
export class CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseModel extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=_links" })
  links?: CodatAssessDataContractsAccountCategoriesAccountCategoriesModelPagedResponseLinksModel;

  @SpeakeasyMetadata({ data: "json, name=pageNumber" })
  pageNumber?: number;

  @SpeakeasyMetadata({ data: "json, name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "json, name=results", elemType: CodatAssessDataContractsAccountCategoriesAccountCategoriesModel })
  results?: CodatAssessDataContractsAccountCategoriesAccountCategoriesModel[];

  @SpeakeasyMetadata({ data: "json, name=totalResults" })
  totalResults?: number;
}
